### DISCUSSION TITLE
Different contracts in different chains? (Tested: didn't work)

### DISCUSSION THREAD

**Discussion by cryptotester** [scaffold-eth/scaffold-eth-2#902] [Q&A] [answered]
I was hoping to be able to us scaffold-eth-2 as a quick and dirty way to debug many of my experimental contract, available on different chains.

I did setup the chains I want to support, e.g. in `scaffold.config.ts` configure support for Arbitrum Sepolia and Mantle Sepolia Testnet.

In `externalContracts.ts` I did setup in the `externalContracts` variable the 2 chains, but:

- On Mantle Sepolia Testnet I want these 2 contract:
  - Foo
  - Bar
- On Arbitrum Sepolia I want to have these 2 completely different contracts:
  - Test1
  - Test2

If I browse on Arbitrum Sepolia to the `/debug` page, I can see Test1 and Test2. All good.

If I switch network to Mantle Sepolia Testnet and go to the `/debug` page, I get this error:

No contract found by the name of "Test1" on chain "Mantle Sepolia Testnet"!
No contract found by the name of "Test2" on chain "Mantle Sepolia Testnet"!

Expected result:

I would expect on mantle testnet to see the contracts configured for mantle testnet: Foo and Bar, instead of the error.

Is this like this by design? Can it be it ready "externalContracts" sorting the chains alphabetically, this even if arbitrum sepolia is configured after mantle sepolia testnet it takes the contracts configured on arbitrum sepolia as "reference" and expects those contracts to be available on all chains?

Looks like a bug to me, if so I will report it as a bug.

**Comment by technophile-04** [comment]
Hey thanks @cryptotester! Actually you should have same contract names on `targetNetworks[0].id` as other chains, as mentioned in docs [note here](https://docs.scaffoldeth.io/hooks/)

![image](https://github.com/user-attachments/assets/0e12dcfa-8904-411b-9a66-612e13d401f8)

Since there are lot of complications getting types autocompletions and functionality working when you have different contracts on different chains checkout[ this for more details](https://github.com/scaffold-eth/scaffold-eth-2/pull/615). So this is an feature not implemented yet but probably in future for sure!

**Reply by cryptotester** [reply]
Thank you, I was thinking it could be like that by design.

So to "cheat" I could simply put the missing contract names and some dummy address for each one of them? I guess it could work? So all the ones I know should fail I won't even debug them and use only the ones I need? I will give it a try. But if this system works I see no reason not to implement such a thing instead of showing the error messages. Maybe adding the support for such a scenario is not that complicated and easily feasible?

**Reply by cryptotester** [reply]
UPDATE: I tested my "cheat" + this change in `ContractUI.tsx` not to show the ugly "No contract found by the name of" error message...

```
  if (!deployedContractData) {
    return (
      <></>
      // <p className="text-3xl mt-14">
      //   {`No contract found by the name of "${contractName}" on chain "${targetNetwork.name}"!`}
      // </p>
    );
  }
```

or better, replace:

```
  if (!deployedContractData) {
    return (
      <p className="text-3xl mt-14">
        {`No contract found by the name of "${contractName}" on chain "${targetNetwork.name}"!`}
      </p>
    );
  }
```

with 
```
  if (!deployedContractData) {
    return (
      <></>
    );
  }
```

This is still a bit suboptimal, as it shows all non deployed contracts in the debug page, but at least if I click them I see that they are empty and I can click and use the deployed ones.

**Comment by ck256-2000** [comment, marked as answer]
Did you update the deployedcongtracts.ts file?  I have similar use case and this worked for me.
